Private bot setting

niansa opened the pull request:
Adds private_bot setting which makes the bot ignore any commands that are not from it's owner if set to True. Returns an error message otherwise.

merlinfuchs committed:
Merge branch 'master' into dev

merlinfuchs committed:
Remove debug print statement

merlinfuchs committed:
Use User instead of Member converter

merlinfuchs committed:
Interval backups are only available for the guild owner

merlinfuchs committed:
minor

merlinfuchs committed:
Fix info command showing team

niansa committed:
Added config option to mark bot as private

Commit:
Implemented private_bot option

Commit:
Added error message and a check if the given message is a command

Commit:
Removed accidental newline

niansa commented:
It's ready for review now ;-)

ThatLifeofGames commented on xenon/bot.py:
```diff
@@ -64,6 +64,10 @@ def __init__(self, *args, **kwargs):
             return
 
+        if message.content.startswith(self.config.prefix) and self.config.private_bot == True and self.config.owner_id != message.author.id:
+            await message.channel.send("Only my owner is allowed to use me!")
```
I hate to sound picky but this doesn't really follow the "standard" of all of xenon's other outputs. Take a look at some of the other outputs, you'll see what I mean :)
Usually everything is in embeds and it's never really in "first person", but this is up to Merlin.

niansa replied:
Alright, let me find another error message... 🤔

niansa replied:
Changed, what do you think?

Commit:
Changed error message

Commit:
More fancy error message

merlinfuchs committed:
Move private bot check into process_commands

This lets discord.py create a context object and check if a command matches the message.

Merlintor commented:
I think it would make sense to create a selfhost branch, because the hosted version of Xenon doesn't benefit in any way of these changes.

niansa commented:
Yeah. But I have no idea how to even do that :thinking:

Merlintor commented:
I created a selfhost branch, you should be able to change your pull request to it.

niansa committed:
Merge branch 'selfhost' into patch-2

niansa commented:
Done
